chore: DLT-2319 update doc headers to title casing

jcarter-dp opened the pull request:
# Chore: DLT 2319 update doc headers to title casing

<!--- Feel free to remove any unused sections -->

## Obligatory GIF (super important!)

<!--- go to giphy.com, pick a gif, click share, copy gif link, paste within the () brackets below. -->

![Obligatory GIF](https://media0.giphy.com/media/v1.Y2lkPTc5MGI3NjExNXRrN2tuaXFyZ2lzcGpkdDE0d3Bob2t0cmh6bm1pdWI4aXQ0OGZ5MyZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/1405fIarqbN7hu/giphy.gif)

## :hammer_and_wrench: Type Of Change

<!--- Tick or place an `x` for the type of change. Should match the type in the commit message / PR title
in https://github.com/dialpad/dialtone/blob/staging/.github/COMMIT_CONVENTION.md -->

These types will increment the version number on release:

- [ ] Fix
- [ ] Feature
- [ ] Performance Improvement
- [ ] Refactor

These types will not increment the version number, but will still deploy to documentation site on release:

- [x] Documentation
- [ ] Build system
- [ ] CI
- [ ] Style (code style changes, not css changes)
- [ ] Test
- [ ] Other (chore)

## :book: Jira Ticket

<!--- Enter the URL of the Jira ticket associated with this PR -->
[DLT-2319](https://dialpad.atlassian.net/browse/DLT-2319)

## :book: Description

<!--- Describe specifically what the changes are -->

## :bulb: Context

<!--- Describe the purpose of the changes -->
<!--- Why did we make these changes? -->
<!--- What problem(s) do they solve? -->

## :pencil: Checklist

<!--- Tick or place an `x` in all of the checkboxes that apply -->
<!--- Remove checkboxes that do not apply -->

For all PRs:

- [x] I have ensured no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).
- [x] I have reviewed my changes.
- [x] I have added all relevant documentation.
- [x] I have considered the performance impact of my change.

For all Vue changes:

- [ ] I have added / updated unit tests.
- [ ] I have made my changes in Vue 2 and Vue 3. Note: you may sync your changes from Vue 2 to Vue 3 (or vice versa) using the `./scripts/dialtone-vue-sync.sh` script. Read docs here: [Dialtone Vue Sync Script](../packages/dialtone-vue3/.github/CONTRIBUTING.md#dialtone-vue-sync-script)
- [ ] I have validated components with a screen reader.
- [ ] I have validated components keyboard navigation.

For all CSS changes:

- [ ] I have used design tokens whenever possible.
- [ ] I have considered how this change will behave on different screen sizes.
- [ ] I have visually validated my change in light and dark mode.
- [ ] I have used gap or flexbox properties for layout instead of margin whenever possible.

If new component:

<!--- There are lots of things to remember when adding a new component to the system! This is so you don't forget any of them. -->

- I am exporting any new components or constants:
  - [ ] from the index.js in the component directory.
  - [ ] from the index.js in the root (either `packages/dialtone-vue2` or `packages/dialtone-vue3`).
- [ ] I have added the styles for the new component to the `packages/dialtone-css` package.
- [ ] I have created a page for the new component on the documentation site in `apps/dialtone-documentation`.
- [ ] I have added the new component to `common/components_list.js`
- [ ] I have created a component story in storybook
- [ ] I have created story / stories for any relevant component variants in storybook
- [ ] I have created a docs page for the component in storybook.
- [ ] I have checked that changing all props/slots via the UI in storybook works as expected.

## :crystal_ball: Next Steps

<!--- Describe any future changes that need to be made after merging the PR, especially any follow up tasks after release. -->

## :camera: Screenshots / GIFs

<!--- Add these if necessary. Since we have deploy previews for every PR it may not always be. -->
<!--- Link any screenshots / GIFs below -->

## :link: Sources

<!--- Add any links to external reference material -->


[DLT-2319]: https://dialpad.atlassian.net/browse/DLT-2319?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

jcarter-dp committed:
Update the dialtone doc

Updating all the markdown headers within the "grammars and mechanics" content page of dialtone's documentation to have proper Title casing.

jcarter-dp committed:
Fix sentence vs title casing link

Under captialization section, fixed the link that properly aligns with sentence vs. title casing section.

jcarter-dp committed:
chore: DLT-2319 update doc titles to title case

Updating all doc pages within dialtone with markdown headers with proper title casing.

jcarter-dp committed:
Update word-wrap file

Update title casing in this file

francisrupert commented:
@jcarter-dp I've changed this to Draft PR. Convert it back once you think you have it all completed.

francisrupert commented:
@jcarter-dp 
You only select one **Type of Fix**. Deselect `Fix`. Should only select `Documentation`.
<img width="772" alt="image" src="https://github.com/user-attachments/assets/823a68a0-b8fd-4a78-8295-518195575183" />

jcarter-dp commented:
@francisrupert Have it ready for review!

francisrupert reviewed (changes requested):
Do all the **Component** pages too, e.g. https://dialtone.dialpad.com/deploy-previews/pr-637/components/avatar.html

jcarter-dp committed:
DLT 2319: Update component pages with proper title casing

Gone through all component pages and updated all headers with proper title casing.

francisrupert commented on apps/dialtone-documentation/docs/components/chip.md:
```diff
@@ -14,7 +14,7 @@ figma_url: https://www.figma.com/file/2adf7JhZOncRyjYiy2joil/DT-Core%3A-Componen
 <!-- <component-combinator component-name="DtChip" /> -->
 
-## Variants and examples
+## Variants And Examples
```
```suggestion
## Variants and Examples
```

francisrupert commented on apps/dialtone-documentation/docs/components/chip.md:
```diff
@@ -96,7 +96,7 @@ vueCode='
 showHtmlWarning />
 
-### With icon and close button
+### With Icon And Close Button
```
```suggestion
### With Icon and Close Button
```

francisrupert commented on apps/dialtone-documentation/docs/components/chip.md:
```diff
@@ -133,7 +133,7 @@ vueCode='
 showHtmlWarning />
 
-### With Avatar and close button
+### With Avatar And Close Button
```
```suggestion
### With Avatar and Close Button
```

francisrupert commented on apps/dialtone-documentation/docs/components/collapsible.md:
```diff
@@ -32,9 +32,7 @@ storybook: https://dialtone.dialpad.com/vue/?path=/story/components-collapsible-
   </div>
 </code-well-header>
 
-The collapsible component consists of two parts: the anchor and the content. If the anchor slot is not filled, the anchor will default to a basic button with text. The content must be provided and is the element that can be hidden or shown when the anchor is clicked.
```
Removed line? "The collapsible component consists..."

francisrupert commented on apps/dialtone-documentation/docs/components/datepicker.md:
```diff
@@ -32,7 +32,7 @@ to the `dt-popover` component.
 With this we accomplish the requirement to have the previous year button focused when the datepicker is opened.
 
-## Variants and examples
+## Variants And Examples
```
```suggestion
## Variants and Examples
```

jcarter-dp committed:
Update headers with title casing in spacing and icons page

Update rest of headers with title casing on spacing and icon pages.

francisrupert commented on apps/dialtone-documentation/docs/components/datepicker.md:
```diff
@@ -407,12 +407,12 @@ In `calendar`:
 
 <!-- TODO: Autogenerate this docs from jsdocs in packages/dialtone-vue2/components/datepicker/formatUtils.js -->
-### formatLong
+### Format Long
```
Don't format prop name.

francisrupert commented on apps/dialtone-documentation/docs/components/datepicker.md:
```diff
@@ -433,7 +433,7 @@ The formatted date string.
 ___
 
-### formatMedium
+### Format Medium
```
Don't format prop name.

francisrupert commented on apps/dialtone-documentation/docs/components/datepicker.md:
```diff
@@ -454,7 +454,7 @@ The formatted date string.
 ___
 
-### formatShort
+### Format Short
```
Don't format prop name.

francisrupert commented on apps/dialtone-documentation/docs/components/datepicker.md:
```diff
@@ -476,7 +476,7 @@ The formatted date string.
 ___
 
-### formatNoYear
+### Format No Year
```
Don't format prop name.

francisrupert commented on apps/dialtone-documentation/docs/components/datepicker.md:
```diff
@@ -498,7 +498,7 @@ The formatted date without the year.
 ___
 
-### formatNumerical
+### Format Numerical
```
Don't format prop name.

francisrupert commented on apps/dialtone-documentation/docs/components/description-list.md:
```diff
@@ -17,7 +17,7 @@ figma: planned
 </code-well-header>
 
-## Variants and examples
+## Variants And Examples
```
`and`

francisrupert commented on apps/dialtone-documentation/docs/components/description-list.md:
```diff
@@ -92,7 +92,7 @@ vueCode='
 />
 
-### With term and description styles
+### With Term And Description Styles
```
`and`

francisrupert commented on apps/dialtone-documentation/docs/components/dropdown.md:
```diff
@@ -62,7 +62,7 @@ Use the Dropdown component when you have a list of links or actions that can be
 - The **list** slot contains a list of items to show as dropdown menu items.
 
-## Variants and examples
+## Variants And Examples
```
`and`

francisrupert commented on apps/dialtone-documentation/docs/components/dropdown.md:
```diff
@@ -137,7 +137,7 @@ vueCode='
 />
 
-### With sections and headings
+### With Sections And Headings
```
`and`

francisrupert commented on apps/dialtone-documentation/docs/components/emoji-picker.md:
```diff
@@ -29,7 +29,7 @@ storybook: https://dialtone.dialpad.com/vue3/?path=/story/components-emoji-picke
 </code-well-header>
 
-## Variants and examples
+## Variants And Examples
```
`and`

francisrupert commented on apps/dialtone-documentation/docs/components/emoji.md:
```diff
@@ -12,9 +12,9 @@ storybook: https://dialtone.dialpad.com/vue/?path=/story/components-emoji--defau
 </code-well-header>
 
-## Variants and examples
+## Variants And Examples
```
`and`

francisrupert commented on apps/dialtone-documentation/docs/components/emoji.md:
```diff
@@ -166,7 +166,7 @@ It follows a similar structure to the [standard emoji json file](https://raw.git
 The required properties are `name`, `category`, `shortname`, `extension`, `custom`.
 
-### Modify data in native emojis
+### Modify Data In Native Emojis
```
`in`

francisrupert commented on apps/dialtone-documentation/docs/components/index.md:
```diff
@@ -1,6 +1,6 @@
 ---
 title: Components
-description: Reusable components solving common UI needs, designed and built to be assembled in countless combinations.
+description: Reusable Components Solving Common UI Needs, Designed and Built to Be Assembled in Countless Combinations.
```
```suggestion
description: Reusable components solving common UI needs, designed and built to be assembled in countless combinations.
```

francisrupert commented on apps/dialtone-documentation/docs/components/popover.md:
```diff
@@ -58,7 +58,7 @@ The content slot will be rendered lazily when the popover is open. By default, t
 - Set the z-index of the dialog to var(--zi-modal-element) if modal, var(--zi-popover) if not.
 
-## Variants and examples
+## Variants And Examples
```
`and`

francisrupert commented on apps/dialtone-documentation/docs/components/presence.md:
```diff
@@ -15,7 +15,7 @@ figma_url: https://www.figma.com/file/2adf7JhZOncRyjYiy2joil/DT-Core%3A-Componen
 Located at the bottom right of an avatar, the `presence` indicator displays a user's current availablity. At a glance, check if a user is available, in a meeting, on a call, away, offline, or set to 'Do Not Disturb'. It's important to note that this indicator is automatically set and cannot be changed manually by the user.
 
-## Variants and examples
+## Variants And Examples
```
`and`

francisrupert reviewed (changes requested):
Almost there.

jcarter-dp committed:
Update tooltip.md

jcarter-dp committed:
Revert "Update word-wrap file"

This reverts commit 00779983a35b7c8db924e9967d1beecb0a9cb607.

jcarter-dp committed:
Fixing propNames and other headers

Fixed the propNames within datepicker.md and updated other headers with proper title casing.

francisrupert committed:
date format demo style

francisrupert reviewed (approved):
Looking good! Great example of a first official PR. Merge away!
